include/gz/sensors/RenderingSensor.hh

```diff
@@ -86,6 +86,10 @@ namespace gz
       /// \param[in] _sensor Sensor to add.
       protected: void AddSensor(rendering::SensorPtr _sensor);
 
+      /// \brief Check if there are any subscribers
+      /// \return True if there are subscribers, false otherwise
+      public: virtual bool HasConnections() const;
```

Comment:
Is there a reason not to put this up on the `Sensor` class instead?

Reply:
I was thinking the same when I took my first glance at this.  I think we could make sensors lazy wherever possible for performance reasons.

Reply:
Thanks! Now we just need to mark all overrides to clean up warnings:

https://build.osrfoundation.org/job/ignition_sensors-ci-pr_any-homebrew-amd64/1113/clang/

Reply:
ok fixed. [d1bcce0](https://github.com/gazebosim/gz-sensors/pull/234/commits/d1bcce047a7e67615f7ac978f7667e3dd328e36d)